src/preamble.js

```diff
@@ -1027,6 +1027,20 @@ function enlargeMemory() {
     }
   }
 
+#if WASM_MEM_MAX != -1
+  // A limit was set for how much we can grow. We should not exceed that
+  // (the wasm binary specifies it, so if we tried, we'd fail anyhow).
```

Comment:
I'm not sure I understand why this is necessary. If WASM_MEM_MAX causes the binary to set the limit, then `reallocBuffer` should fail and we'd fail down on line 1051? But from the linked issue it looks like we are not, so maybe something else is wrong?

Reply:
Yeah, maybe this should be explained better. The issue is if we are 64MB and the max is 100MB, then if we normally double the size - so we'd try to 64->128MB, which fails. Instead, with this PR, we'll try for 64->100MB, which might work.